app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -111,6 +117,19 @@ class FxaRepo(
         return accountManager.beginAuthenticationAsync()
     }
 
+    @SuppressLint("CheckResult") // This survives for the duration of the app
+    private fun setupTelemetry() {
+        accountState
+            // Filter out intermediate states. E.g., when signing in, we see 'NotAuthenticated',
+            // then 'AuthenticatedWithProfile' and 'AuthenticatedNoProfile' in quick succession. We
+            // only want to use the final value here
+            .debounce(10, TimeUnit.SECONDS)
+            .map { it is NeedsReauthentication }
+            .subscribe {
+                telemetryIntegration.fxaNeedsReauthentication(it)
```

Comment:
nit: most telemetry methods are called `*Event`: should we do that here too?

Also, it's not really `fxaNeedsReauthentication[Event]`: it's `doesFxaNeedReauthenticationEvent` or similar.